flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/TimestampedValue.java

```diff
@@ -18,28 +18,77 @@
 package org.apache.flink.streaming.runtime.operators.windowing;
 
 import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
 
 /**
  * Stores the value and the timestamp of the record.
+ * 
  * @param <T> The type encapsulated value
  */
 @PublicEvolving
 public class TimestampedValue<T> {
 
+	/** The actual value held by this record */
 	private T value;
+
+	/** The timestamp of the record */
 	private long timestamp;
 
+	/** Flag whether the timestamp is actually set */
+	private boolean hasTimestamp;
+
+	/**
+	 * Creates a new TimestampedValue. The record does not have a timestamp.
+	 */
+	public TimestampedValue(T value) {
```

Comment:
This constructor is never called because in `EvictingWindowOperator` you always instantiate `TimestampedValue` with a value and a timestamp even though the `StreamRecord` might not have a timestamp.